### PR TITLE
Add ability to redirect logging

### PR DESCRIPTION
**Pull request by shawnburke**
The logger is very noisy with other unit tests.  Add the ability to change:

```
WithLogWriter(myOtherWriter)
```
or suppress

```
WithLogWriter(nil)
```

Logging output.

### CODE REVIEW

File: duct.go [hunk 1]
```diff
@@ -138,6 +156,12 @@ func (c *Composer) GetNetworkID() string {
 	return c.netID
 }
 
+type nullWriter struct{}
```
Comment (erikh) [inline duct.go:159]: wouldn't `ioutil.Discard` suitably work here? do we need our own type?

File: duct.go [hunk 2]
```diff
@@ -324,7 +356,7 @@ func (c *Composer) Teardown(ctx context.Context) error {
 
 			log.Printf("Removing container: [%s]", cont.Name)
 			if err := client.RemoveContainer(dc.RemoveContainerOptions{ID: cont.id, Force: true, Context: ctx}); err != nil {
-				log.Println(err)
+				log.Printf("Error shutting down continer: [%s] %v\n", cont.Name, err)
```
Comment (erikh) [inline duct.go:359]: you misspelled container, also please do not add newlines to log messages, they gum up the output.

File: duct_test.go [hunk 3]
```diff
@@ -335,3 +336,39 @@ func TestWithExistingNetwork(t *testing.T) {
...
+	if err := c.Teardown(context.Background()); err != nil {
+		t.Fatal(err)
+	}
+
+	logs := buffer.String()
+	t.Log(logs)
+	if len(logs) == 0 {
+		t.Fatal("Didn't capture logs")
+	}
+}
```
Comment (erikh) [inline duct_test.go:374]: great test!

**Review by erikh — CHANGES_REQUESTED** [review]
so, minor changes aside the feature looks good -- thanks for contributing it!

I would like you to cycle back with the fixes requested then I will validate locally; no CI yet. I'll get to that soonish.

File: duct.go (hunk 1 above)
Comment (shawnburke) [reply duct.go:159]: oh i was looking for that, happy to replace thanks

File: duct.go (hunk 2 above)
Comment (shawnburke) [reply duct.go:359]: good catch, ya forgot this wasn't regular printf.

**Comment by shawnburke** [comment]
Yes, seems like it would be pretty easy to set up travis-ci for this project :)

Cool, no big hurry, I'm consuming my fork for now.

**Comment by erikh** [comment]
Thanks for your contribution! I'll review the other one now.

**Pull request merged**